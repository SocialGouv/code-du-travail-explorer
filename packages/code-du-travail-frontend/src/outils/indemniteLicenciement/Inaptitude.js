import React from "react";
import PropTypes from "prop-types";
import { Container, Section } from "@cdt/ui";

import { PrevNextStepper } from "./PrevNextStepper";
import { Label, RadioContainer } from "./stepStyles";

class Inaptitude extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.bool,
    onPrevious: PropTypes.func,
    onNext: PropTypes.func,
    nextDisabled: PropTypes.bool
  };
  static defaultProps = {
    value: false,
    nextDisabled: false
  };

  render() {
    const { onChange, value, onNext, onPrevious, nextDisabled } = this.props;
    return (
      <React.Fragment>
        <Section light>
          <h2>
            Le licenciement a t-il fait il suite au prononc√© d&apos;une
            inaptitude d&apos;origine professionnelle ( accident du travail ou
            maladie professionnelle)&nbsp;?
          </h2>
          <RadioContainer>
            <Label>
              <input
                type="radio"
                onChange={() => onChange(true)}
                name="inaptitude"
                checked={value === true}
              />
              <span>Oui</span>
            </Label>
            <Label>
              <input
                type="radio"
                onChange={() => onChange(false)}
                name="inaptitude"
                checked={value === false}
              />
              <span>Non</span>
            </Label>
          </RadioContainer>
        </Section>
        <Container>
          <PrevNextStepper
            onPrev={onPrevious}
            onNext={onNext}
            nextDisabled={nextDisabled}
          />
        </Container>
      </React.Fragment>
    );
  }
}

export { Inaptitude };
