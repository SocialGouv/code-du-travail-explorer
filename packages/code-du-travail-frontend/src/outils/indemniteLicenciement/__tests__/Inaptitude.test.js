import React from "react";
import { render, fireEvent } from "react-testing-library";
import { Inaptitude } from "../Inaptitude";

describe("<Inaptitude />", () => {
  it("should render", () => {
    const onChange = jest.fn();
    const value = false;
    const { container } = render(
      <Inaptitude value={value} onChange={onChange} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should call onChange", () => {
    const onChange = jest.fn();
    const value = false;
    const { getByLabelText } = render(
      <Inaptitude value={value} onChange={onChange} />
    );

    const input = getByLabelText(/oui/i);
    fireEvent.click(input);

    expect(onChange).toBeCalledWith(true);
  });
});
