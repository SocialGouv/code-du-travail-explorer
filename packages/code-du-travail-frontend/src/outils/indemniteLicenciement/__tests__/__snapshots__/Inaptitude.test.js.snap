// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inaptitude /> should render 1`] = `
<div>
  <section
    class="section-light"
  >
    <div
      class="container"
    >
      <div
        class="wrapper-light"
      >
        <h2>
          Êtes-vous licencié pour inaptitude suite à accident du travail ou maladie professionnelle reconnue ?
        </h2>
        <div
          class="sc-ifAKCX dmUlWY"
        >
          <label
            class="sc-bxivhb dTOYdh"
          >
            <input
              name="inaptitude"
              type="radio"
            />
            <span>
              Oui
            </span>
          </label>
          <label
            class="sc-bxivhb dTOYdh"
          >
            <input
              checked=""
              name="inaptitude"
              type="radio"
            />
            <span>
              Non
            </span>
          </label>
        </div>
      </div>
    </div>
  </section>
  <div
    class="container"
  >
    <div
      style="display: flex; flex-direction: row; margin-top: 20px; font-size: 1rem;"
    >
      <div />
    </div>
  </div>
</div>
`;
